tdms/tests/unit/test_hdf5_io.cpp

```diff
@@ -5,35 +5,47 @@
 #include "hdf5_io.h"
 
 // std
+#include <cstdlib>
 #include <ctime>
 #include <filesystem>
 #include <random>
+#include <string>
 
 // external
 #include <catch2/catch_approx.hpp>
 #include <catch2/catch_test_macros.hpp>
 #include <spdlog/spdlog.h>
 
 // tdms
+#include "arrays.h"
+#include "fdtd_grid_initialiser.h"
 #include "unit_test_utils.h"
 
-using tdms_tests::create_tmp_dir;// unit_test_utils.h
+using tdms_tests::create_tmp_dir;
+
+TEST_CASE("Wrong datatype passed to ijk.") {
+  auto tmp = create_tmp_dir();
+  HDF5Writer r(tmp.string() + "/why.h5");
+  // r.data_dump("x");
+}
 
 TEST_CASE("Test file I/O construction/destruction.") {
-  // setup - temporary directory
+
+  // test-case wide setup - temporary directory
   auto tmp = create_tmp_dir();
 
   SECTION("Check file creation.") {
     HDF5Writer f(tmp.string() + "/test_file_constructor.h5");
     CHECK(f.is_ok());
-  }// destructor called as we leave scope
+  }
```

Comment:
```suggestion
  }// Destructor called as we leave scope
```
IMHO it's a useful comment to keep.